tests/assert-jobs/reporter-otlp/main.go

```diff
@@ -0,0 +1,242 @@
+
+	logrus.Debugln("Using", reportingProtocol, "to report the traces")
+
+	var driver otlp.ProtocolDriver
+
+	switch reportingProtocol {
+	case "grpc":
+		driver = otlpgrpc.NewDriver(
+			otlpgrpc.WithInsecure(),
+			otlpgrpc.WithEndpoint(collector),
+			otlpgrpc.WithDialOption(grpc.WithBlock()),
+		)
+	case "http":
+		driver = otlphttp.NewDriver(
+			otlphttp.WithInsecure(),
+			otlphttp.WithEndpoint(collector),
+		)
+	default:
+		logrus.Fatalln("Reporting protocol", reportingProtocol, "not recognized")
+	}
+
+	exp, err := otlp.NewExporter(ctx, driver)
+	if err != nil {
+		logrus.Fatalln("error creating the exporter", err)
+	}
+
+	res, err := resource.New(ctx,
+		resource.WithAttributes(
+			semconv.ServiceNameKey.String(serviceName),
+		),
+	)
+	if err != nil {
+		logrus.Fatalln("error creating the resource", err)
+	}
+
+	bsp := sdktrace.NewBatchSpanProcessor(exp)
+	tracerProvider := sdktrace.NewTracerProvider(
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
+		sdktrace.WithResource(res),
+		sdktrace.WithSpanProcessor(bsp),
+	)
+
+	cont := controller.New(
+		processor.New(
+			simple.NewWithExactDistribution(),
+			exp,
+		),
+		controller.WithExporter(exp),
+		controller.WithCollectPeriod(2*time.Second),
+	)
+
+	otel.SetTextMapPropagator(propagation.TraceContext{})
+	otel.SetTracerProvider(tracerProvider)
+
+	err = cont.Start(context.Background())
+	if err != nil {
+		logrus.Fatalln("error while starting the controller", err)
+	}
+
+	// This function should be called when the tracing features will not be
+	// used anymore
+	return func() {
+		err1 := cont.Stop(context.Background())
+		err2 := tracerProvider.Shutdown(ctx)
+
+		// The errors are checked later to try to run all the "closing" tasks
+		if err1 != nil {
+			logrus.Fatalln("failed while stopping the controller", err1)
+		}
+		if err2 != nil {
+			logrus.Fatalln("failed to shutting down the tracer provider", err2)
+		}
+	}
+}
+
+// Generate substans inside a span
+// ctx: context for the program
+// depth: how many spans should be created as child spans of this one
+func generateSubSpans(ctx context.Context, depth int) {
+	if depth == 0 {
+		return
+	}
+	tracer := otel.Tracer(tracerName)
+	ctx, span := tracer.Start(ctx, fmt.Sprintf("subspan-%d", depth))
+	defer span.End()
+	logrus.Debugln("\tGenerating subspan", depth)
+	time.Sleep(time.Millisecond * 30)
+	generateSubSpans(ctx, depth-1)
+}
+
+// Generate some traces
+// jaegerEnpoint: where Jaeger endpoint is located
+// serviceName: service to use for the span
+// operationName: operation described in the span
+// traces: number of traces to generate
+func generateTraces(jaegerEndpoint string, serviceName string, operationName string, traces int) {
+	logrus.Debugln("Trying to generate traces")
+	shutdown := initProvider(serviceName)
+	defer shutdown()
+
+	tracer := otel.Tracer(tracerName)
+
+	logrus.Debugln("Generating traces!")
+
+	i := 0
+	for {
+		logrus.Debugf("Generating trace %d/%d", i, traces)
```

Comment:
Isnt this loop now running for ever? since traces is never used. We only place it into this log line?

Reply:
So its intended to run for ever?

Reply:
The KUTTL timeout will stop it when needed